packages/react/src/componentsV2/baseCheckout/index.tsx

```diff
@@ -0,0 +1,201 @@
+  totalAmount,
+  PricingComponent,
+}: BaseCheckoutProps) => {
+  const {
+    currencyList,
+    paymentDetails,
+    tokenSwapLoading,
+    tokenSwapQuote,
+    tokenSwapError,
+    removeTokenSwapError,
+  } = useHelioProvider();
+
+  const { HelioSDK } = useCompositionRoot();
+
+  const [normalizedPrice, setNormalizedPrice] = useState(0);
+  const [activeCurrency, setActiveCurrency] = useState<Currency | null>(null);
+  const [showSwapMenu, setShowSwapMenu] = useState(false);
+  const [showQRCode, setShowQRCode] = useState(false);
+
+  const canSelectCurrency = allowedCurrencies.length > 1;
+
+  const payButtonText =
+    showSwapMenu && tokenSwapQuote?.from?.symbol && !tokenSwapError
+      ? `PAY IN ${tokenSwapQuote.from.symbol}`
+      : 'PAY';
+  const payButtonDisable =
+    (showSwapMenu && !!tokenSwapError) || tokenSwapLoading;
+
+  const initialValues = getInitialValues(
+    paymentDetails,
+    normalizedPrice,
+    canSelectCurrency,
+    paymentDetails?.dynamic
+      ? allowedCurrencies?.[0].symbol
+      : paymentDetails?.currency?.symbol
+  );
+
+  useEffect(() => {
+    if (allowedCurrencies.length === 1) {
+      setActiveCurrency(allowedCurrencies[0]);
+    } else if (!canSelectCurrency) {
+      setActiveCurrency(
+        getCurrency(currencyList, paymentDetails?.currency?.symbol)
+      );
+    }
+  }, [paymentDetails?.currency, canSelectCurrency]);
+
+  useEffect(() => {
+    if (
+      paymentDetails?.currency != null &&
+      paymentDetails?.normalizedPrice != null
+    ) {
+      setNormalizedPrice(
+        HelioSDK.tokenConversionService.convertFromMinimalUnits(
+          paymentDetails?.currency?.symbol,
+          paymentDetails?.normalizedPrice
+        )
+      );
+    }
+  }, [paymentDetails]);
+
+  useEffect(() => {
+    if (!showSwapMenu) {
+      removeTokenSwapError();
+    }
+  }, [showSwapMenu]);
+
+  return createPortal(
+    <StyledBaseCheckoutWrapper>
+      <StyledBaseCheckoutContainer>
+        <CheckoutHeader
+          icon={
+            activeCurrency && (
+              <CurrencyIcon gradient iconName={activeCurrency?.symbol || ''} />
+            )
+          }
+          title={activeCurrency ? `Pay with ${activeCurrency?.symbol}` : 'Pay'}
+          showSwap={paymentDetails?.features.canSwapTokens}
+          isSwapShown={showSwapMenu}
+          toggleSwap={() => setShowSwapMenu(!showSwapMenu)}
+          onHide={onHide}
+          showQRCode={showQRCode}
+        />
+
+        <StyledBaseCheckoutBody>
+          {showQRCode && (
+            <div>
+              {!paymentDetails?.features.canChangePrice && (
+                <PriceBanner
+                  title="Total price:"
+                  amount={formatTotalPrice(totalAmount || normalizedPrice)}
+                  currency={activeCurrency?.symbol}
+                />
+              )}
+              <QRCodeCard
+                phantomDeepLink={`https://phantom.app/ul/browse/${HelioSDK.configService.getPaymentFullLink(
```

Comment:
Phantom deeplinks usually have a ref param, please check the docs 
![image](https://user-images.githubusercontent.com/56583386/219648902-3588b4a9-649e-48cf-8938-c4c97f374a9c.png)
This is how it looks on dash. Also check to see if this actually works